probe-rs/src/debug/unit_info.rs

```diff
@@ -988,53 +987,37 @@ impl UnitInfo {
 
                 // This needs to resolve the pointer before the regular recursion can continue.
                 match node.entry().attr(gimli::DW_AT_type) {
-                    Ok(Some(data_type_attribute)) => {
-                        match data_type_attribute.value() {
-                            gimli::AttributeValue::UnitRef(unit_ref) => {
-                                // The default behaviour is to defer the processing of child types.
-                                child_variable.variable_node_type =
-                                    VariableNodeType::ReferenceOffset(unit_ref);
-                                if let VariableType::Pointer(optional_name) =
-                                    &child_variable.type_name
-                                {
-                                    #[allow(clippy::unwrap_used)]
-                                    // Use of `unwrap` below is safe because we first check for `is_none()`.
-                                    if optional_name.is_none()
```

Comment:
I'm conflicted whether this is a mistake. The `data_ptr` in slice types is a pointer to a `u8` but the pointer does not have a `DW_AT_type`, but otherwise Rust tends to name its pointer types (i.e. I see `*const u8` in other places in DWARF explorer).

Reply:
I see what you mean. I looked at what is generated and I'm OK with the way it shows up now. If we get feedback, we can always improve later :)